### ISSUE TITLE
Using synthesized instruments with the extended soundbank crashes the game

### ISSUE THREAD

**Issue by intheshed34** [coop-deluxe/sm64coopdx#108]
**Describe the bug**
Using synthesized instruments with the extended soundbank crashes the game.

**To Reproduce**
1. Setting any track in a .m64 file to use a synthesized instrument with the extended soundbank (Instruments 145 Saw Wave or 146 Square Wave) crashes the game

**Expected behavior**
The song should be able to play with synthesized instruments with the extended soundbank just fine without crashing.

**Screenshots**

First crash screen with the first track using instrument 145 Saw Wave
![extended_synth_crash_0](https://github.com/coop-deluxe/sm64coopdx/assets/98070839/2b378fc1-d5e8-48dc-a4d6-4780649eedb3)

Second crash screen with a different .m64, the first track using instrument 145 Saw Wave
![extended_synth_crash_1](https://github.com/coop-deluxe/sm64coopdx/assets/98070839/65421c40-1a4c-4d98-9e6d-88bb4ba5f2c0)

Third crash screen with same .m64 as above, only using instrument 146 Square Wave
![extended_synth_crash_2](https://github.com/coop-deluxe/sm64coopdx/assets/98070839/71ac1dc6-5ba7-4eb5-9e5d-de4fe8e9fae1)

**Desktop:**
 - OS: Windows 10 Home
 - Browser Firefox
 - Version 22H2

**Additional context**
This crash is also present on sm64ex-coop and just to be sure; I used the Metal Cap
soundbank which has a Saw Wave instrument with my second .m64, and that worked just perfectly fine.

**Comment by Flower35** [comment]
Hi. I don't know the exact workings of seq64, but it looks like that Extended Soundbanks (instrument IDs ≥ 128) are not supported in `sm64ex-coop` / `sm64coopdx`.

https://github.com/coop-deluxe/sm64coopdx/blob/30ccad30a8be27e3f0fa34fd2ab27303a65eb4f4/src/audio/internal.h#L490-L491

https://github.com/coop-deluxe/sm64coopdx/blob/30ccad30a8be27e3f0fa34fd2ab27303a65eb4f4/src/audio/playback.c#L1135-L1143

---

Indices `0x80-0x83 (128-131)` are reserved for `gWaveSamples[4]` (synthetic waves), and indices `0x84-0xFF (132-255)` are invalid and will crash the game.

  * In your case, `145 Saw Wave` was translated to `gWaveSamples[17]`, and `146 Square Wave` was translated to `gWaveSamples[18]`, both of which point outside the expected bounds of a static array.

  * Fun fact: data pointed to by these unsupported instrument IDs is the `sSawtoothWave`, but those bytes (on your screenshots: `RAX: DC01D801D401D001` and `RAX: EC01E801E401E001`) cannot be interpreted as pointers.

**Comment by intheshed34** [comment]
Shoot! It seems like you're right, I went back and used other instrument IDs greater than 131, and it seems like using those also crash the game (My mistake for not further testing this on other instruments).

I also went ahead and checked the indices `0x80-0x83 (128-131)`, and those were the correct synth instruments like you said.

I guess I'll close this issue now (Plus the one I made to the ex-coop repo) despite the extended soundbank file listing off instrument IDs 0-203, still weird that synths are improperly mislabeled.
